model/host/host.go

```diff
@@ -1677,36 +1683,46 @@ func StaleRunningTaskIDs(staleness time.Duration) ([]task.Task, error) {
 	return out, err
 }
 
-// AddTags adds the specified tags to the host document, or modifies
-// an existing tag if it can be modified.
-func (h *Host) AddTags(tags []Tag) {
-	for _, new := range tags {
-		found := false
-		for i, old := range h.InstanceTags {
-			if old.Key == new.Key {
-				found = true
-				if old.CanBeModified {
-					h.InstanceTags[i] = new
-				}
-				break
+func (h *Host) addTag(new Tag, hasPermissions bool) {
```

Comment:
Do you need to validate that a duplicate tag isn't added?

Reply:
Not sure what you mean -- I think this method does this (if the key matches the key of a tag already on the host, then it updates that tag if it can be modified and otherwise returns) but I could be missing something.